src/components/DataCollector/index.js

```diff
@@ -45,6 +48,15 @@ const createDataCollector = ({ eventManager }) => {
         event.userXdm = xdm;
         event.userData = data;
 
+        const warnings = event.validate();
+        if (warnings.length) {
+          logger.warn(
+            `Invalid event command options:\n\t - ${warnings.join(
+              "\n\t - "
+            )}\nFor documentation covering the event command see: ${CONFIG_DOC_URI}`
```

Comment:
This seems brittle. We're not really validating the user input here but instead are asking the event whether it's valid. Whether the event reports itself as valid may not always be correlated to the options the user passed in through this command; we're making the assumption that it is related, but it might not be. Right now the two might match (when options are valid, the event is valid and vice versa), but if someone were to change the event.validate logic to incorporate something other than the options the user provided through this command, that assumption of correlation would break. Since we're reporting on the validity of the user input here, I'd rather we validate the user input rather than validate that the event object is valid. I think that makes even more sense since we want to use config validators for command options eventually (soon, hopefully).

Reply:
Logged here: https://jira.corp.adobe.com/browse/CORE-39143